internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -44,6 +46,9 @@ type Receiver struct {
 var _ component.MetricsReceiver = (*Receiver)(nil)
 
 func NewReceiver(logger *zap.Logger, config Config, nextConsumer consumer.MetricsConsumer) *Receiver {
+	logrus.SetOutput(ioutil.Discard)
+	logrus.AddHook(&ZapHooks{ZapLogger: logger})
```

Comment:
If you have multiple instances of a smartagent receiver (not uncommon, especially when they're started dynamically with receiver_creator) if a receiver starts and stops it's still going to be registered with logrus. Also log messages will be duplicated because it's a global. So if you start 3 smartagent receivers I think the same log message would be logged 3 times. Unsure what the solution is.

Reply:
Hey @jrcamp I did some refactoring. Please have a look and comment.

Reply:
I'm not sure how this helps the scenarios above. I don't really see a solution to making the two fit without modifying the smart agent code to have a per-monitor instance logger.

Reply:
I think each monitor does use its own logger for the most part (std w/ identifying fields), but are scoped to their corresponding `Configure()` calls, which may be too late for purposes.  I think that a complete solution would be a good end goal, and imo should be investigated so we don't punt off the bat.  It would likely require an additional monitor method to return a wrapped logger ensuring we can access it before the monitor has started.

A short term stopgap could be to have a singleton smart agent receiver logger that is always used, overwriting the component name field of the first registered receiver logger.

Reply:
Please explain "if a receiver starts and stops it's still going to be registered with logrus" a bit further. So newing a smartagent receiver hooks the supplied zap logger to the global logrus logger for redirection. zap logger instances are hooked only once throughout the life of the application. Different zap logger instances that write to the same destination will cause duplication.

Reply:
> I think each monitor does use its own logger for the most part (std w/ identifying fields), but are scoped to their corresponding `Configure()` calls, which may be too late for purposes. I think that a complete solution would be a good end goal, and imo should be investigated so we don't punt off the bat. It would likely require an additional monitor method to return a wrapped logger ensuring we can access it before the monitor has started.
> 
> A short term stopgap could be to have a singleton smart agent receiver logger that is always used, overwriting the component name field of the first registered receiver logger.

Wait, I only see monitors using the exported std logger!

Reply:
Right, it's the same underlying logger, but the monitors use their own wrapper with fields.